metricflow/sql_clients/databricks.py

```diff
@@ -42,35 +52,36 @@ class DatabricksEngineAttributes(SqlEngineAttributes):
 class DatabricksSqlClient(BaseSqlClientImplementation):
     """Client used to connect to Databricks engine."""
 
-    def __init__(self, host: str, http_path: str, access_token: str) -> None:
+    def __init__(self, host: str, http_path: str, access_token: str) -> None:  # noqa: D
         self.host = host
         self.http_path = http_path
         self.access_token = access_token
 
+    # TODO: why do we even need this method? just for tests? Shouldn't we use the same configs in tests as prod?
     @staticmethod
     def from_connection_details(url: str, password: Optional[str]) -> DatabricksSqlClient:  # noqa: D
         # Is the input format right for this? What's normal for databricks users?
         try:
-            url, http_path = url.split(";")  # TODO: there might not only be http path in there
-            parsed_url = sqlalchemy.engine.url.make_url(url)
-        except:  # which exceptions to swallow?
-            # What type of exception to raise?
-            raise ValueError("Unexpected url format. Expected: `databricks://<HOST>:443;HttpPath=<HTTP PATH>")
-
-        dialect = SqlDialect.DATABRICKS.value
-        if parsed_url.drivername != dialect:
-            raise ValueError(f"Expected dialect '{dialect}' in {url}")
+            split_url = url.split(";")  # TODO: there might not only be http path in there
+            parsed_url = sqlalchemy.engine.url.make_url(split_url[0])
+            for piece in split_url:
+                if HTTP_PATH_KEY in piece.lower():
+                    __, http_path = piece.split("=")
+            dialect = SqlDialect.DATABRICKS.value
+            if not http_path or parsed_url.drivername != dialect or not parsed_url.host:
+                raise ValueError
+        except:  # noqa: E722
```

Comment:
Nit: Instead of swallowing the exception, can you do `raise ValueError(...) from e`? You would have to re-work throwing that value error though.